tests/test_single_topology_combined.py

```diff
@@ -0,0 +1,390 @@
+        # 2a) lambda offset and plane parameters
+        combined_lambda_offset_idxs = combined_host_guest_nonbonded.get_lambda_offset_idxs()
+        combined_lambda_plane_idxs = combined_host_guest_nonbonded.get_lambda_plane_idxs()
+
+        for a_idx, (offset_idx, plane_idx) in enumerate(zip(combined_lambda_offset_idxs, combined_lambda_plane_idxs)):
+            if a_idx < num_host_atoms:
+                assert offset_idx == 0
+                assert plane_idx == 0
+            else:
+                # guest atom
+                guest_atom_idx = a_idx - num_host_atoms
+                indicator = st3.c_flags[guest_atom_idx]
+                if indicator == 0:
+                    # core
+                    assert offset_idx == 0
+                    assert plane_idx == 0
+                elif indicator == 1:
+                    # mol_a dummy
+                    assert offset_idx == 1
+                    assert plane_idx == 0
+                elif indicator == 2:
+                    # mol_b dummy
+                    assert offset_idx == 1
+                    assert plane_idx == -1
+                else:
+                    assert 0
+
+        # 2b) nonbonded parameter interpolation checks
+        mol_a_charges = st3.ff.q_handle.parameterize(st3.mol_a)
+        mol_a_sig_eps = st3.ff.lj_handle.parameterize(st3.mol_a)
+
+        mol_b_charges = st3.ff.q_handle.parameterize(st3.mol_b)
+        mol_b_sig_eps = st3.ff.lj_handle.parameterize(st3.mol_b)
+
+        c_to_a = dict()
+        for k, v in enumerate(st3.a_to_c):
+            c_to_a[v] = k
+
+        c_to_b = dict()
+        for k, v in enumerate(st3.b_to_c):
+            c_to_b[v] = k
+
+        for a_idx, (test_q, test_sig, test_eps) in enumerate(combined_host_guest_nonbonded.params):
+
+            if a_idx < num_host_atoms:
+                continue
+
+            guest_atom_idx = a_idx - num_host_atoms
+            indicator = st3.c_flags[guest_atom_idx]
+
+            # dummy atom qlj parameters are arbitrary (since they will be decoupled via lambda parameters)
+            if indicator != 0:
+                continue
+
+            if lamb == 0.0:
+                # should resemble mol_a at lambda=0
+                ref_q = mol_a_charges[c_to_a[guest_atom_idx]]
+                ref_sig, ref_eps = mol_a_sig_eps[c_to_a[guest_atom_idx]]
+
+                assert ref_q == test_q
+                assert test_sig == ref_sig
+                assert test_eps == ref_eps
+
+            elif lamb == 1.0:
+                # should resemble mol_b at lambda=1
+                ref_q = mol_b_charges[c_to_b[guest_atom_idx]]
+                ref_sig, ref_eps = mol_b_sig_eps[c_to_b[guest_atom_idx]]
+
+                assert ref_q == test_q
+                assert test_sig == ref_sig
+                assert test_eps == ref_eps
+
+
+@pytest.mark.skip(reason="This is currently too slow to run on CI")
+def test_combine_host_guest():
```

Comment:
leaving this here for now for when we setup more extensive system tests later on

Reply:
Would it make sense to run this as a nightly test?